File: code/back-office/server/src/main/java/internship/project/election/dto/ApiResponse.java
package internship.project.election.dto;

import java.util.Map;

import lombok.Data;

@Data
public class ApiResponse {
    private Object payload;
    private String message;
    private Map<String, String> errors;

    public ApiResponse() {
    }

    public ApiResponse(Object payload, String message) {
        this.payload = payload;
        this.message = message;
    }

    public ApiResponse(Object payload, String message, Map<String, String> errors) {
        this.payload = payload;
        this.message = message;
        this.errors = errors;
    }
}

File: code/back-office/front/src/app/dto/api.response.ts
export class ApiResponse<P> {
  payload?: P;
  message?: string;
  errors?: { [key: string]: string[] };

  constructor(
    payload?: P,
    message?: string,
    errors?: { [key: string]: string[] }
  ) {
    this.payload = payload;
    this.message = message;
    this.errors = errors;
  }
}
